File: data/readable_format/cpp/file1118.cpp
#include <iostream>
#include <cassert>
#include <algorithm> // For std::swap

// Declaration of the sswap function
void sswap(int n, float sx[], int incx, float sy[], int incy);

int main() {
    float sx[5] = {1.0, 2.0, 3.0, 4.0, 5.0};
    float sy[5] = {5.0, 4.0, 3.0, 2.0, 1.0};
    
    // Call the sswap function
    sswap(5, sx, 1, sy, 1);
    
    // Expected arrays after swap
    float expected_sx[5] = {5.0, 4.0, 3.0, 2.0, 1.0};
    float expected_sy[5] = {1.0, 2.0, 3.0, 4.0, 5.0};
    
    // Verify the results
    for (int i = 0; i < 5; ++i) {
        assert(sx[i] == expected_sx[i] && "sx array did not match the expected result.");
        assert(sy[i] == expected_sy[i] && "sy array did not match the expected result.");
    }
    
    std::cout << "Test passed successfully." << std::endl;

    return 0;
}

// Definition of the sswap function
void sswap(int n, float sx[], int incx, float sy[], int incy) {
    if (n <= 0) return;

    if (incx == 1 && incy == 1) {
        // Special case for unit strides
        int m = n % 3;
        if (m != 0) {
            for (int i = 0; i < m; ++i) {
                std::swap(sx[i], sy[i]);
            }
        }
        if (n < 3) return;

        for (int i = m; i < n; i += 3) {
            std::swap(sx[i], sy[i]);
            std::swap(sx[i + 1], sy[i + 1]);
            std::swap(sx[i + 2], sy[i + 2]);
        }
    } else {
        // General case for arbitrary strides
        int ix = 0;
        int iy = 0;

        if (incx < 0) ix = (n - 1) * incx;
        if (incy < 0) iy = (n - 1) * incy;

        for (int i = 0; i < n; ++i) {
            std::swap(sx[ix], sy[iy]);
            ix += incx;
            iy += incy;
        }
    }
}

File: data/readable_format/fortran/file1118.f90
program test_sswap
  implicit none
  real :: sx(5) = [1.0, 2.0, 3.0, 4.0, 5.0]
  real :: sy(5) = [5.0, 4.0, 3.0, 2.0, 1.0]
  integer :: i

  call sswap(5, sx, 1, sy, 1)
  
  print *, 'sx after swap: '
  do i=1,5
     print *, sx(i)
  end do
  
  print *, 'sy after swap: '
  do i=1,5
     print *, sy(i)
  end do

contains

  subroutine sswap(n, sx, incx, sy, incy)
    real, dimension(*) :: sx, sy
    real :: stemp
    integer :: i, incx, incy, ix, iy, m, mp1, n

    if (n <= 0) return
    if (incx == 1 .and. incy == 1) then
      m = mod(n, 3)
      if (m /= 0) then
        do i = 1, m
          stemp = sx(i)
          sx(i) = sy(i)
          sy(i) = stemp
        end do
      end if
      if (n < 3) return
      mp1 = m + 1
      do i = mp1, n, 3
        stemp = sx(i)
        sx(i) = sy(i)
        sy(i) = stemp
        stemp = sx(i + 1)
        sx(i + 1) = sy(i + 1)
        sy(i + 1) = stemp
        stemp = sx(i + 2)
        sx(i + 2) = sy(i + 2)
        sy(i + 2) = stemp
      end do
    else
      ix = 1
      iy = 1
      if (incx < 0) ix = (-n + 1) * incx + 1
      if (incy < 0) iy = (-n + 1) * incy + 1
      do i = 1, n
        stemp = sx(ix)
        sx(ix) = sy(iy)
        sy(iy) = stemp
        ix = ix + incx
        iy = iy + incy
      end do
    end if
  end subroutine sswap

end program test_sswap